cored/users-csv
# PoC: Users CSV import

## Summary

1. Enhanced the `UploadCsv` service to handle edge cases in CSV parsing and validate user data more robustly.
2. Implemented error handling to gracefully manage and notify users of issues with malformed or invalid CSV files.

### Biggest Problem

The biggest issue addressed by this pull request was the need for robust error handling and improved validation when processing CSV uploads. 

## Solution

To address these issues comprehensively, the solution focuses on enhancing the `UploadCsv` service to:
- Validate each row of the CSV file against specified criteria for user data integrity.
- Handle exceptions and errors gracefully, providing clear feedback to users about any issues encountered during CSV processing.

## How

- Implement more stringent checks for CSV format validity before processing begins.
- Utilize custom validators and error handling mechanisms within the service to ensure that each user is properly validated

### Follow up

What would happen if the CSV is too big? 
Consider triggering the process with a job 
Consider implementing pagination? 
UX/UI improvements

## Steps to QA

1. Upload various CSV files containing valid and invalid user data.
2. Verify that the application correctly identifies and reports errors for invalid data entries.
3. Check that the application processes valid entries without errors and saves them to the database.

## Definition of Done

- [x] All new or changed code is covered with unit/integration tests to ensure functionality.
